src/osd/ReplicatedPG.cc

```diff
@@ -3605,7 +3605,7 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 
         op.extent.length = total_read;
 
-        ::encode_destructively(bl, osd_op.outdata);
+        osd_op.outdata.claim_append(bl);
```

Comment:
@yuyuyu101 this change is good.

we don't have a `::encode_destructively()` for `std::map`, and the idea of `::encode_destructively()` does not apply for non-buffer types, i guess.